File: src/basic programming/time comparison/protein_counter.go
package main

import (
	"fmt"
	"log"
	"os"
	"time"
)

func main() {
	start := time.Now()
	file, err := os.ReadFile("data.dat")
	if err != nil {
		panic(err)
	}
	A, C, T, G := 0, 0, 0, 0
	for i := 0; i < len(file); i++ {
		if file[i] == 65 {
			A += 1
		} else if file[i] == 67 {
			C += 1
		} else if file[i] == 71 {
			G += 1
		} else if file[i] == 84 {
			T += 1
		}
	}
	fmt.Printf("A = %d\nC = %d\nG = %d\nT = %d\n", A, C, G, T)
	elapsed := time.Since(start)

	log.Printf("The counter took %s", elapsed)
}

File: src/basic programming/time comparison/protein_counter.py
import time
start_time = time.time()
file = open("data.dat").readlines()
file = file[0]

char = [protein for protein in file]

A = C = G = T = 0
for i in range(len(char)):
  if char[i] == "A":
    A = A + 1
  elif char[i] == "C":
    C = C + 1
  elif char[i] == "G":
    G = G + 1
  elif char[i] == "T":
    T = T + 1
print("A = {}\nC = {}\nG = {}\nT = {}".format(A, C, G, T))
print("The program took {}us to execute".format((time.time() - start_time)*(10**6)))

